iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/Substations.java

```diff
@@ -33,7 +34,8 @@ private Substations() {
                 return false;
             }
             for (String tag : geographicalTags) {
-                if (!substation.getGeographicalTags().contains(tag)) {
+                String geographicalTagsStr = substation.getProperty("geographicalTags");
+                if (geographicalTagsStr != null && !Arrays.asList(geographicalTagsStr.split(",")).contains(tag)) {
                     return false;
                 }
             }
```

Comment:
geographicalTagsStr (and the split) can be outside the loop. But anyway the loop itself might be clearer if replaced by:
`Arrays.asList(geographicalTagsStr.split(",")).containsAll(Arrays.asList(geographicalTags))`